platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/DefaultInternalEventValidator.java

```diff
@@ -157,16 +171,59 @@ public DefaultInternalEventValidator(
      * @return true if the required fields of the event are non-null, otherwise false
      */
     private boolean areRequiredFieldsNonNull(@NonNull final PlatformEvent event) {
-        if (event.getSignature() == null) {
-            // do not log the event itself, since toString would throw a NullPointerException
-            nullUnhashedDataLogger.error(EXCEPTION.getMarker(), "Event has null signature");
-            nullUnhashedDataAccumulator.update(1);
+        final GossipEvent gossipEvent = event.getGossipEvent();
+        final EventCore eventCore = gossipEvent.eventCore();
+        if (eventCore == null
+                || eventCore.timeCreated() == null
+                || eventCore.version() == null
+                || eventCore.parents().stream().anyMatch(Objects::isNull)
+                || gossipEvent.eventTransaction().stream().anyMatch(DefaultInternalEventValidator::isTransactionNull)) {
+            nullFieldLogger.error(EXCEPTION.getMarker(), "Event has null field {}", gossipEvent);
+            nullFieldAccumulator.update(1);
             return false;
         }
 
         return true;
     }
 
+    /**
+     * Checks whether the transaction is null.
+     * @param transaction the transaction to check
+     * @return true if the transaction is null, otherwise false
+     */
+    private static boolean isTransactionNull(@Nullable final EventTransaction transaction) {
+        return transaction == null
+                || transaction.transaction() == null
+                || transaction.transaction().value() == null;
+    }
+
+    /**
+     * Checks whether the {@link com.hedera.pbj.runtime.io.buffer.Bytes} fields of an event are the expected length.
+     *
+     * @param event the event to check
+     * @return true if the byte fields of the event are the correct length, otherwise false
+     */
+    private boolean areByteFieldsCorrectLength(@NonNull final PlatformEvent event) {
+        final GossipEvent gossipEvent = event.getGossipEvent();
+        final EventCore eventCore = event.getEventCore();
+        if (gossipEvent.signature().length() != SignatureType.RSA.signatureLength()) {
```

Comment:
Here and several other places in this class are lacking coverage. Suggest using intellij coverage tool to address

Reply:
i added unit tests for this

Reply:
it now has coverage for all new code